internal/engine/wazevo/backend/isa/arm64/lower_instr.go

```diff
@@ -427,7 +429,7 @@ func (m *machine) lowerFpuUniOp(op fpuUniOp, in, out ssa.Value) {
 	m.insert(neg)
 }
 
-func (m *machine) lowerFpuToInt(rd, rn operand, ctx regalloc.VReg, signed, src64bit, dst64bit bool) {
+func (m *machine) lowerFpuToInt(rd, rn operand, ctx regalloc.VReg, signed, src64bit, dst64bit, nonTrapping bool) {
 	// First of all, we have to clear the FPU flags.
```

Comment:
I believe this shouldn't be neede when non trapping

Reply:
that's ok, I only copied it because the original compiler is doing it

https://github.com/tetratelabs/wazero/blob/ea634990050bbe9af5ce230cce9811d54566c60f/internal/engine/compiler/impl_arm64.go#L2189-L2192

at this point I'd rather create a `lowerFpuToIntNontrapping()` since it becomes three lines. WDYT? or maybe add `lowerFpuToIntTrapping()` which invokes `lowerFpuToInt()` that becomes the 3-liner:


```
func lowerFpuToIntTrapping(...) {
   // clear flags
   lowerFpuToInt(...)
   // handle trap
}
func lowerFpuToInt(...) {
	cvt := m.allocateInstr()
	cvt.asFpuToInt(rd, rn, signed, src64bit, dst64bit)
	m.insert(cvt)
}
```

Reply:
up to you!

Reply:
either works for me 🍷

Reply:
I am keeping it simple (just if statements) because I guess I am still too jet-lagged to refactor it right :D basically the ctx register becomes useless for the nontrapping version, but that requires a few more changes to bubble up the chain and something was going wrong at the regalloc stage. I can refactor it later.

Reply:
I am marking this as ready, but if you'd rather me do the refactor I can do push it here later